core/src/main/java/org/eclipse/hono/util/RequestResponseResult.java

```diff
@@ -103,6 +103,15 @@ public final CacheDirective getCacheDirective() {
         }
     }
 
+    /**
+     * Gets access read-only access to the response message's <em>application-properties</em>.
+     * 
+     * @return The unmodifiable map of the application properties. Never returns {@code null}.
+     */
+    public Map<String, Object> getApplicationProperties() {
```

Comment:
FMPOV this should be final as all the other accessors as there is no obvious reason for it to be overridden, is there?